File: src/assets/request_headers.py
from pydantic import BaseModel
from typing import Optional

class CodeRequest(BaseModel):
    language: str
    code: str


class CommandRequest(BaseModel):
    command: str
    id: str
    fetch_manual: Optional[bool] = True
    server_name: Optional[str] = None
    language: Optional[str] = None
    code: Optional[str] = None

File: src_ts/assets/request_headers.ts
import { z } from "zod";

export const CommandRequestSchema = z.object({
  command: z.string(),
  id: z.string(),
  fetch_manual: z.boolean().optional().default(true),
  server_name: z.string().optional(),
  language: z.string().optional(),
  code: z.string().optional(),
});

export type CommandRequest = z.infer<typeof CommandRequestSchema>;

export type OrchestratorResponse =
  | {
      status: number;
      result: string | null;
      info?: string;
      id: string;
    }
  | {
      status: number;
      detail: string;
      id: string;
    };

